### PR TITLE
Replace BGzipCodec with equivalent BGZFCodec from Hadoop-BAM

### PR DESCRIPTION
**Pull request by tomwhite**
The idea is to consolidate on the BGZF code in Hadoop-BAM that other projects (like GATK, ADAM) use too. It's not ready for merge yet as the write side (and support for the .bgz suffix) is not in Hadoop-BAM yet: https://github.com/HadoopGenomics/Hadoop-BAM/pull/106.

### CODE REVIEW

**Comment by tomwhite** [comment]
I've updated this with the code from Hadoop-BAM (in 7.6.0), so it's ready for review. Can you take a look please @cseed?

**Comment by cseed** [comment]
I'll take a look.  Thanks Tom!

File: src/main/scala/org/broadinstitute/hail/Utils.scala [hunk 1]
```diff
@@ -967,23 +968,19 @@ object Utils extends Logging {
     val outputStream = destFS.create(destPath)
 
     try {
-      var i = 0
-      while (i < srcFileStatuses.length) {
-        val fileStatus = srcFileStatuses(i)
-        val lenAdjust: Long = if (isBGzip && i < srcFileStatuses.length - 1)
-          -28
-        else
-          0
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/Utils.scala]: Tom, I just want to double-check the BGZFCodec doesn't write empty gzip blocks at the end of files by default, for example, when compressing the output of saveAsTextFile.  While it is BGZ compliant to have empty blocks, we found several tools that incorrectly treated them as premature end of files.

Comment (tomwhite) [inline src/main/scala/org/broadinstitute/hail/Utils.scala]: @cseed, no, BGZFCodec doesn't write empty gzip blocks at the end of files. See 

https://github.com/HadoopGenomics/Hadoop-BAM/blob/master/src/main/java/org/seqdoop/hadoop_bam/util/BGZFCompressionOutputStream.java#L44

**Comment by cseed** [comment]
jenkins

**Pull request merged**